test/api/activity_types_api_test.rb

```diff
@@ -50,4 +50,50 @@ def test_put_activity_types
     first_activity_type = ActivityType.first
     assert_json_matches_model(last_response_body, first_activity_type, response_keys)
   end
+
+  def test_post_activity_types_cannot_auth
+    # Number of Activity type before post new activity type
+    number_of_activity_type = ActivityType.count
+
+    # A user with student role which does not have premision to post a activity type
+    user = FactoryBot.create(:user, :student)
+
+    # Create a dummy activity type
+    data_to_post = {
+      activity_type: FactoryBot.build(:activity_type),
+      auth_token: auth_token
+    }
+
+    # Perform POST, but the student user does not have permissions to post it.
+    post_json '/api/activity_types', with_auth_token(data_to_post, user)
+
+    # Check if the post does not get through
+    assert_equal 403, last_response.status  
+
+    # Check if the number of activity type is the same as initially
+    assert_equal ActivityType.count, number_of_activity_type
+  end  
+
+  def test_put_activity_types_cannot_auth
+    # Number of Activity type before put new activity type
+    number_of_activity_type = ActivityType.count
+
+    # A user with student role which does not have premision to put a activity type
+    user = FactoryBot.build(:user, :student)
+
+    # Create a dummy activity type
+    data_to_put = {
+      activity_type: FactoryBot.build(:activity_type),
+      auth_token: auth_token
+    }
+
+    # Update activity_type with id = 1, but the student user does not have permissions to put it.
+    put_json '/api/activity_types/1', with_auth_token(data_to_put, user)
```

Comment:
Please do not put `data_to_put` into the activity type of which id is explicitly `1`. Let create a new activity type by FactoryBot `activity_type = FactoryBot.create(:activity_type)` then assign `data_to_put`into to replace that newly created `activity_type` of which id can be called by `activity_type.id`. If this does not make sense to you, please ask